src/main/java/org/assertj/core/internal/Iterables.java

```diff
@@ -1083,7 +1083,7 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
     if (!diff.differencesFound()) {
       // actual and values have the same elements but are they in the same order ?
       int i = 0;
-      for (Object elementFromActual : actual) {
+      for (Object elementFromActual : actualAsList) {
```

Comment:
don't reuse `actualAsList`, although it's not the case it could have modified in `diff`, simply create another list

Reply:
I'm not sure about this suggestion because I don't believe we should be encouraging methods with side-effects. Also I note that there is an explicit comment on the IterableDiff class that reads "immutable" so, are you sure you want a second copy of the data?

Reply:
fair enough